File: Data Structures & Algorithms/house-robber-ii/submission-0.java
class Solution {
    
    public int rob(int[] nums) {
        return Math.max(nums[0], Math.max(helper(Arrays.copyOfRange(nums, 1, nums.length)), helper(Arrays.copyOfRange(nums, 0, nums.length - 1))));
    }

    private int helper(int[] nums) {
        int rob1 = 0, rob2 = 0;

        for (int n : nums) {
            int newRob = Math.max(rob1 + n, rob2);
            rob1 = rob2;
            rob2 = newRob;
        }
        return rob2;
    }
}

File: Data Structures & Algorithms/max-consecutive-ones/submission-0.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        max = 0
        curr = 0
        for i in nums:
            if i == 1:
                curr = curr + 1
            else:
                curr = 0

            if max < curr:
                max = curr
            
        return max